watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1556,6 +1690,170 @@ def eq_dl_thickness(self, mem, side, t, x):
+            doc="The mass diffusivity of the solute as molecules (not individual ions)",
+        )
+        self.hydraulic_diameter = Var(initialize=1e-3, units=pyunits.meter)
+        self.Re = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Reynolds Number"
+        )
+        self.Sc = Var(
+            initialize=2000, units=pyunits.dimensionless, doc="Schmidt Number"
+        )
+        self.Sh = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Sherwood Number"
+        )
+
+        if self.config.hydraulic_diameter_method == HydraulicDiameterMethod.defined:
+            _log.warning("Do not forget to FIX the channel hydraulic diameter in [m]!")
+        else:
+
+            @self.Constraint(
+                doc="To calculate hydraulic diameter",
+            )
+            def eq_hydraulic_diameter(self):
+                if (
+                    self.config.hydraulic_diameter_method
+                    == HydraulicDiameterMethod.conventional
+                ):
+                    return (
+                        self.hydraulic_diameter
+                        == 2
+                        * self.channel_height
+                        * self.cell_width
+                        * self.spacer_porosity
+                        * (self.channel_height + self.cell_width) ** -1
+                    )
+                else:
+                    self.spacer_specific_area = Var(
+                        initialize=1e4, units=pyunits.meter**-1
+                    )
+                    return (
+                        self.hydraulic_diameter
+                        == 4
+                        * self.spacer_porosity
+                        * (
+                            2 * self.channel_height**-1
+                            + (1 - self.spacer_porosity) * self.spacer_specific_area
+                        )
+                        ** -1
+                    )
+
+        @self.Constraint(
+            doc="To calculate Re",
+        )
+        def eq_Re(self):
+
+            return (
+                self.Re
+                == self.dens_mass
+                * self.velocity_D[0, 0]
+                * self.hydraulic_diameter
+                * self.visc_d**-1
+            )
+
+        @self.Constraint(
+            doc="To calculate Sc",
+        )
+        def eq_Sc(self):
+
+            return (
+                self.Sc
+                == self.visc_d
+                * self.dens_mass**-1
+                * self.general_mass_diffusivity**-1
+            )
+
+        @self.Constraint(
+            doc="To calculate Sc",
+        )
+        def eq_Sh(self):
+
+            return self.Sh == 0.29 * self.Re**0.5 * self.Sc**0.33
+
+    def _pressure_drop_calculation(self):
+        self.pressure_drop = Var(
+            self.flowsheet().time,
+            initialize=1e4,
+            units=pyunits.pascal * pyunits.meter**-1,
+            doc="pressure drop per unit of length in a single ED channel",
```

Comment:
So, you're using a ControlVolume, which should have a config option called `has_pressure_change`. When set True, a `deltaP` variable is created. Although this may work, and might even be clearer, for the sake of consistency, you probably should be using the ControlVolume to do this (e.g., as RO model does).

Reply:
Thanks! Just worked this out... I can push the new adds to this PR or in my next PR.